app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx

```diff
@@ -106,6 +158,42 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
     )
   }
 
+  async fetchInitialKeys() {
+    const publicKeys = await getPublicKeys()
+    this.setState({
+      publicKeys,
+      loadingPublicKeys: false,
+      address: {
+        value: publicKeys[0].key,
+        label: this.unencodedHexToAddress(publicKeys[0].key)
+      }
+    })
+  }
+
+  async fetchAdditionalKeys() {
+    const { publicKeys } = this.state
+    this.setState({ loadingPublicKeys: true })
+    const lastAccountLoaded = publicKeys[publicKeys.length - 1].account
+    const nextBatchOfKeys = await getPublicKeys(lastAccountLoaded)
+    this.setState(state => ({
+      publicKeys: [...state.publicKeys, ...nextBatchOfKeys],
+      loadingPublicKeys: false
+    }))
+  }
+
+  unencodedHexToAddress = (hexString: string) => {
+    const encodedKey = wallet.getPublicKeyEncoded(hexString)
+    return new wallet.Account(encodedKey).address
+  }
+
+  createOptionsFromKeys = () => {
+    const options = this.state.publicKeys.map(key => ({
```

Comment:
I think `key.key` is a bit weird, how about if you call it `publicKey`, since you are mapping `publicKeys`?